src/API/Authentication.php

```diff
@@ -409,6 +410,10 @@ public function code_exchange(string $code, string $redirect_uri) : array
             'grant_type' => 'authorization_code',
         ];
 
+        if ($code_verifier) {
+            $options += ['code_verifier' => $code_verifier];
```

Comment:
For the sake of simplicity:

```suggestion
            $options['code_verifier'] = $code_verifier;
```